src/components/capture-area/CaptureAreaSearchValue/CaptureAreaSearchValue.test.ts

```diff
@@ -0,0 +1,103 @@
+import { describe, it, expect } from 'vitest';
+import { ref } from 'vue';
+import { shallowMount } from '@vue/test-utils';
+import CaptureAreaSearchValueVue from '@/components/capture-area/CaptureAreaSearchValue/CaptureAreaSearchValue.vue';
+import RegexTextField from '@/components/capture-area/RegexTextField/RegexTextField.vue';
+import { Vigad } from '@/proc/Vigad';
+import {
+    defaultValueRegexSettings,
+    defaultAfterConstraintsSettings,
+    defaultBeforeConstraintsSettings,
+} from '@/components/capture-area/CaptureAreaSearchValue/ResetSettings';
+
+describe('CaptureAreaSearchValue', () => {
+    /**
+     * Vigad instance
+     */
+    const vigad = ref(Vigad.getInstance());
+
+    /**
+     * Add capture area
+     */
+    vigad.value.addCaptureArea(100, 100, 0, 0);
+
+    /**
+     * Capture areas
+     */
+    const captureAreas = ref(vigad.value.getAllCaptureAreas());
+
+    /**
+     * Value regex
+     */
+    const valueRegex = vigad.value
+        .getCaptureArea(captureAreas.value[0].getId())
+        .getRegexGroups()[0]
+        .getValueRegex();
+
+    /**
+     * Before constraint
+     */
+    const beforeConstraint = vigad.value
+        .getCaptureArea(captureAreas.value[0].getId())
+        .getRegexGroups()[0]
+        .getConstraintRegex()[0];
+
+    /**
+     * After constraint
+     */
+    const afterConstraint = vigad.value
+        .getCaptureArea(captureAreas.value[0].getId())
+        .getRegexGroups()[0]
+        .getConstraintRegex()[1];
+
+    const props = {
+        captureAreaId: captureAreas.value[0].getId(),
+    };
+
+    it('render with RegexTextField with proper props', () => {
+        const wrapper = shallowMount(CaptureAreaSearchValueVue, {
+            props: props,
+        });
+
+        expect(wrapper.findAllComponents(RegexTextField)).toHaveLength(3);
+        expect(
+            wrapper.findComponent({
+                name: 'RegexTextField',
+                props: {
+                    label: 'Search value',
+                    placeholder: 'Enter search value',
+                    prependIcon: 'mdi-table-column',
+                    captureAreaId: captureAreas.value[0].getId(),
+                    regex: valueRegex,
+                    resetOptions: defaultValueRegexSettings,
+                },
+            })
+        ).toBeTruthy();
```

Comment:
This schema seems kind of odd to me... What is actually being checked here?

Reply:
So here I check if the component actually renders 3 of the custom RegexText Fields, and check if it really finds the Value Regex, Before Constraint and After Constraint Text Field with the right properties etc.